test/IntegrationTests/StartupHookTests.cs

```diff
@@ -101,6 +104,48 @@ public void ApplicationIsNotIncluded()
 #endif
     }
 
+    [Fact]
+    [Trait("Category", "EndToEnd")]
+    public void SubmitMetrics()
+    {
+        SetEnvironmentVariable("OTEL_DOTNET_AUTO_METRICS_ADDITIONAL_SOURCES", "MyCompany.MyProduct.MyLibrary");
+        var collectorPort = TcpPortProvider.GetOpenPort();
+        using var collector = new MockCollector(Output, collectorPort);
+
+        const int expectedMetricRequests = 1;
+
+        var testSettings = new TestSettings
+        {
+            MetricsSettings = new MetricsSettings { Port = collectorPort },
+            EnableStartupHook = true,
+            Arguments = "-m"
+        };
+
+        using var processResult = RunTestApplicationAndWaitForExit(testSettings);
+        Assert.True(processResult.ExitCode >= 0, $"Process exited with code {processResult.ExitCode} and exception: {processResult.StandardError}");
+        var metricRequests = collector.WaitForMetrics(expectedMetricRequests, TimeSpan.FromSeconds(5));
+
+        using (new AssertionScope())
+        {
+            metricRequests.Count.Should().Be(expectedMetricRequests);
+
+            var resourceMetrics = metricRequests.Single().ResourceMetrics.Single();
```

Comment:
using `Single()` can cause a crash with an ugly exception. it maybe even hard to know which of the `Single()` has crashed from the stack trace

As far as I remember FluentAssertions has something like `collection.Should().ContainSingle().Which...`

reference: https://fluentassertions.com/collections/